crates/turborepo-lib/src/shim.rs

```diff
@@ -337,16 +337,17 @@ impl LocalTurboState {
     // - `pnpm install`
     // - `npm install --install-strategy=linked`
     fn generate_linked_path(root_path: &AbsoluteSystemPath) -> Option<AbsoluteSystemPathBuf> {
-        let canonical_path = fs_canonicalize(
-            root_path
-                .as_path()
-                .join("node_modules")
-                .join("turbo")
-                .join(".."),
-        )
-        .ok()?;
-
-        AbsoluteSystemPathBuf::try_from(canonical_path).ok()
+        // root_path/node_modules/turbo is a symlink. Canonicalize the symlink to what
+        // it points to. We do this _before_ traversing up to the parent,
+        // because on Windows, if you canonicalize a path that ends with `/..`
+        // it traverses to the parent directory before it follows the symlink,
+        // leading to the wrong place. We could separate the Windows
+        // implementation, but this workaround works for other platforms as
+        // well.
+        let canonical_path =
+            fs_canonicalize(root_path.as_path().join("node_modules").join("turbo")).ok()?;
+
+        AbsoluteSystemPathBuf::try_from(canonical_path.join("..")).ok()
```

Comment:
If you have tried this, totally ignore the suggested code change, but maybe add a comment that warns that `foo/bar/.. != foo/` in this case.
```suggestion
        AbsoluteSystemPathBuf::try_from(canonical_path.parent()?).ok()
```